lang/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerExceptionsTest.kt

```diff
@@ -143,7 +143,7 @@ class EvaluatingCompilerExceptionsTest : EvaluatorTestBase() {
     fun missingAlias() =
         // Same query as previous test--but DO NOT throw exception this time because of UndefinedVariableBehavior.MISSING
         runEvaluatorTestCase(
-            sqlWithUndefinedVariable, expectedResult = "[null]",
+            sqlWithUndefinedVariable, expectedResult = "$BAG_ANNOTATION::[$MISSING_ANNOTATION::null]",
```

Comment:
Is it fair to say that all test adjustments amounted to adding one of these two annotations to expected ion values -- that is, there were absolutely no other reasons tests had to be adjusted as the result of changing `ION_WITHOUT_BAG_AND_MISSING_ANNOTATIONS` to `STRICT`?  
Whatever the answer, this is something to share in the PR description.

Reply:
I can *imagine* reasons for `ION` to remain in some cases, but the question [in the review preamble, to which you seem to be responding here] is about the specific tests touched by this PR.  Can you *point to* a reason why these tests should be `ION` and not `STRICT`?

Reply:
The bug of `ExprValue` iterator has not been addressed yet, https://github.com/partiql/partiql-lang-kotlin/issues/833

Reply:
I don't see this through yet.  Is there a specific test here for which #833 bug plays out?  Does it play out in *all* the tests if they use `STRICT` rather than `ION`?

Reply:
This one, https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerFromSourceByTests.kt#L75

Reply:
Also, another question is that if we are querying against an Ion file or an `IonValue`, is the output supposed to be `IonValue` or `ExprValue`?

Reply:
I think it should be `IonValue` in this case. Namely, if the binding data is in format of ION, the output should be in format of ION, since we are querying against ION.

Reply:
If so, the `rangeOverListNested` test that you are pointing at can be at `ION`, while the majority still switches to `STRICT` -- the resting framework has a parameter on the `runEvaluatorTestCase` that can do that, from what I see.  
The reason why this test gets special treatment is worth documenting, with reference to #833. 

> Also, another question is that if we are querying against an Ion file or an IonValue, ....

I don't see how this reasoning is valid: in `PipelineEvaluatorTestAdapter`, even in the `ION` format, the actual value still surfaces as an ExprValue and then gets converted to Ion, just for the purpose of comparing to the expected (specified as Ion).  Isn't the whole purpose of `STRICT` format you introduced to avoid this round-about?

Reply:
I mean one feature of PartiQL is to query against Ion files. In this case, users expect to get the result back in format of Ion. So if we are writing integration test to test this feature, I think the input & output should be Ion, instead of `ExprValue`, since we might not property transform `ExprValue` to `IonValue`.

Reply:
If you are querying against a JSON file, you hope the query engine returns a JSON value. And if you are querying an ION file, you hope it returns an Ion value to you, right?

Reply:
Right now we are talking about the tests touched by this PR.  Do you say your arguments still apply?

Reply:
I agree that we should use `STRICT` for all the test cases eventually. But my approach in this PR is that, we change `ION_WITHOUT_ANNOTATION` to `ION` first, and then let's see if we should switch to `STRICT` next. And probably, for some test cases, we are not able to do so. So we should figure this out in the next PRs. But in this PR, let's just remove `ION_WITHOUT_ANNOTATION`, and switch to `ION`, since this change for any test case always makes sense.

Reply:
If you think we should look at each test case and try to switch it to `STRICT` just in this PR, I can do that.

Reply:
Yes.   

Here is my initial suggestion that started this conversation

> Since STRICT is the ultimate goal, you could be done and done with these tests, but now you'll have to plow through them again, now also commingled with all the other ION tests!

This pointed out that going straight to `STRICT` would've been a more efficient way of proceeding, since you (and I) would have looked at most of these tests once rather than twice.
If you really want to look at these tests again when eliminating `ION`, so be it.  

I still want to see an improved PR description, since that goes into the git commit.  Other comments in this PR have some of the suggestions on what to document.